operator/pkg/helmreconciler/prune.go

```diff
@@ -207,9 +214,27 @@ func (h *HelmReconciler) GetPrunedResources(revision string, includeClusterResou
 		}
 		usList = append(usList, objects)
 	}
+
 	return usList, nil
 }
 
+// getIstioOperatorCR is a helper function to delete IstioOperator CR during purge,
+// otherwise the resources would be reconciled back later if there is in-cluster operator deployment.
+// And it is needed to remove the IstioOperator CRD.
+func (h *HelmReconciler) getIstioOperatorCR() *unstructured.UnstructuredList {
+	objects := &unstructured.UnstructuredList{}
+	objects.SetGroupVersionKind(schema.GroupVersionKind{Group: "install.istio.io", Version: "v1alpha1", Kind: "IstioOperator"})
+	if err := h.client.List(context.TODO(), objects); err == nil {
+		for _, obj := range objects.Items {
+			obj.SetFinalizers([]string{})
+			if err := h.client.Patch(context.TODO(), &obj, client.Merge); err != nil {
```

Comment:
I don't think something in `GetPrunedResources` should be mutating an object, this seems wrong to me.

Reply:
separated the patch part to delete function